Major Network backend changes (Testing Purposes Only)

BaronGreenback opened the pull request:
A staged rollout of this PR can be found here.

https://github.com/jellyfin/jellyfin/pull/4124
https://github.com/jellyfin/jellyfin/pull/4125
#4404 
#4138 
#4137 
#4136 
https://github.com/jellyfin/jellyfin-web/pull/1979

Config Settings:
=====
(Value in brackets is the default setting.)

**LocalNetworkSubnets** (empty) : Defines local subnets, and is used by the DLNA subsystem.
If blank, all the subnets of any private interfaces will be assumed to be the LAN.
Interface names, descriptions and subnets can be specified, as well as exclusions (addresses beginning with !)
usage:
10.0.0.0/8,!192.168.1.1 means 10.0.0.0/8 is in the LAN, but ignore traffic to/from 192.168.1.1
!eth1 means every interface but exclude the interface eth1.

**LocalNetworkAddresses** (empty) : lists the interfaces that jellyfin binds to. Default is all interfaces.
Interface names, descriptions and subnets can be specified, as well as exclusions (addresses starting with !)
usage:
10.10.10.10 means listen only on 10.10.10.10
!eth1 means listen on every interface except eth1.

The system intelligently selects the most appropriate interface, so it is possible to specify both internally and externally facing network adapters.

dlna.xml advanced settings: (direct xml edit required)
====

**EnablePlayToTracing** (false) : If dlna debugging is enabled in conjunction to this, ssdp packet tracing is enabled. Warning: This has performance implications.

**AutoCreatePlayToProfiles** (false) : Auto creates user DLNA profiles for customisation.

system.xml advanced settings: (direct xml edit required)
====

**HDHomerunPortRange** (Random Port) : Specify a UDP port, or range of UDP ports that HDHomerun uses.

**GatewayMonitorPeriod** (60 seconds) : How often the gateway is pinged to ensure it's operational.

**EnableIPV6** (False) : Disable/enable IPv6 usage.

**EnableIPV4** (True) : Disable/enable IPv4 usage. (Useful for IP6 testing).

**AutoDiscovery** (true) : Enables/Disables network autodiscovery over udp port 7359. 

**AutoDiscoveryTracing** (false) : Enables tracing on jellyfin's autodiscovery.

**UDPPortRange** (Any) : Port range to use for internal UDP ports. Can be fixed, or in a range. eg. 1024-2048

**TrustAllIP6Interfaces** (false) ; All IPv6 interfaces are deemed to be local. (See conversation at #3515.)

**EnableMultiSocketBinding** (true) : System will use specific ip bound sockets. Autocalculated, but can be overridden.

**UPnPCreateHttpPortMap** (false): Maps the HTTP port in addition to the HTTPS port on any network devices detected via uPNP (if active). Historically, uPnP has mapped both the HTTP and HTTP ports inbound. This is now no longer the case.

**UDPSendCount** (2) : Sets the number of times each SSDP message is transmitted. 

**UDPSendDelay** (100) : Sets the delay between each group of SSDP messages. (in ms)

**EnableSSDPTracing** (false) : If dlna debugging is enabled in conjunction to this, ssdp packet tracing is enabled. Warning: This has performance implications.

**SSDPTracingFilter** (empty) : Used in conjunction with EnableSSDPTracing. Set to an IP address to filter logs to a specific address.

**IgnoreVirtualInterfaces** (true): Ignores any interface that matches those in **VirtualInterfaceNames**.

**VirtualInterfaceNames** (vEthernet*) : The name of the virtual interfaces to ignore.

**PublishedServerUriBySubnet** (empty) : list of [subnets|interfaces]/[serverUri|host|ip] [port] values that replicate the --published-server-url command line argument, but on a per subnet basis, enabling.
usage:

The following mimics the --published-server-url option but only for external requests.

```
<PublishedServerUriBySubnet>
<string>external=http://myjellyfin.org</string>
</PublishedServerUriBySubnet>
```

The following causes any requests from the subnet defined by eth1 to respond with a hostname jellyfin.com
```
<PublishedServerUriBySubnet>
<string>eth1=jellyfin.com </string>
</PublishedServerUriBySubnet>
```

The following causes any requests from the subnet defined by eth1 to respond with a hostname jellyfin.com, and everything else to return the address http://jellyexternal.com
```
<PublishedServerUriBySubnet>
<string>eth1=jellyfin.com</string>
<string>remaining=http://jellyexternal.com</string>
</PublishedServerUriBySubnet>
```

The following changes the url address dependant upon source request.
```
<PublishedServerUriBySubnet>
<string>192.168.1.0/24=net1access:1270</string>
<string>192.168.2.1/24=net2access</string>
<string>external=jellyfinoutside</string>
</PublishedServerUriBySubnet> 
```

Logging
====

To enable debug logging, edit logging.default.json, and add

```
    "Serilog": {
        "MinimumLevel": {
            "Override": {
	       "Jellyfin.Networking" : "Debug",
               "Mono.Nat.Upnp.UpnpSearcher ": "Debug",
               "Emby.Dlna" : "Debug"
```

Retired settings
=====

**BlastAliveMessages** - no longer valid. Try increasing UDPSendCount, if the server is not detected reliably by other UPnP devices on your network. However, be aware that this may exponentially increase network traffic.

**SendOnlyMatchedHost** (false)- No longer valid. It causes SSDP replies to be sent to all subnets, not just the local ones.


Docker image crobibero/jellyfin:3257

BaronGreenback commented:
Previous was https://github.com/jellyfin/jellyfin/pull/3257.

My github got screwed up, so this is a manual compare with the latest Unstable (26/8/2020)

BaronGreenback commented:
https://trac.videolan.org/vlc/ticket/25055#comment:1

VLC does not support host names within the location field of dlna.

BaronGreenback commented:
Suggestion:

Instead of using --published-server-url

Add the following to the system,xml
<PublishedServerUriBySubnet>
<string>external=**published-server-url**</string>
</PublishedServerUriBySubnet> 

This will only us change the url to the **published-server-url** for requests coming from outside the LAN.

crobibero commented:
Updated `crobibero/jellyfin:3257` to 4902546
Image tag will be kept from previous PR for easier updating.

PrplHaz4 commented:
Looks like I lost NOTIFY messages on 4902546 again.

Also, seeing this (HOST has double 1900) in M-SEARCH now:

```
M-SEARCH * HTTP/1.1
HOST: 239.255.255.250:1900:1900
USER-AGENT: UPnP/1.0 DLNADOC/1.50 Platinum/1.0.4.2 /af9f06335f4d490685987391b5f6b4be
MAN: "ssdp:discover"
ST: ssdp:all
MX: 3
01-NLS: 8e4c66ec-5733-4f15-9a65-e93c6d57ec26
BOOTID.UPNP.ORG: 8e4c66ec-5733-4f15-9a65-e93c6d57ec26
OPT: "http://schemas.upnp.org/upnp/1/0/"; ns=01
SERVER: Linux/Unix 4.14.138.0 UPnP/1.0 RSSDP/1.0
CONFIGID.UPNP.ORG: 1
```

BaronGreenback commented:
Removed the extra :1900 - don't know where that came from.

The only other change is the additional fields for SSDP that are supposed to be optional for IPv4.

01-NLS: 8e4c66ec-5733-4f15-9a65-e93c6d57ec26
BOOTID.UPNP.ORG: 8e4c66ec-5733-4f15-9a65-e93c6d57ec26
OPT: "http://schemas.upnp.org/upnp/1/0/"; ns=01
CONFIGID.UPNP.ORG: 1

What client are you using to connect?

PrplHaz4 commented:
> 
> What client are you using to connect?

This time it was just the UPnP Device Sniffer - I tried to confirm with Wireshark, and found these NOTIFY messages coming in....guess my first stop before opening my mouth should be Wireshark:

```
    NOTIFY * HTTP/1.1\r\n
    HOST: 239.255.255.250:1900\r\n
    DATE: Thu, 27 Aug 2020 20:31:26 GMT\r\n
    CACHE-CONTROL: max-age = 1800\r\n
    LOCATION: https://jftest2.mydomain.com/dlna/af9f06335f4d490685987391b5f6b4be/description.xml\r\n
    NTS: ssdp:alive\r\n
    NT: urn:schemas-upnp-org:device:MediaServer:1\r\n
    USN: uuid:af9f06335f4d490685987391b5f6b4be::urn:schemas-upnp-org:device:MediaServer:1\r\n
    01-NLS: 8e4c66ec-5733-4f15-9a65-e93c6d57ec26\r\n
    BOOTID.UPNP.ORG: 8e4c66ec-5733-4f15-9a65-e93c6d57ec26\r\n
    OPT: "http://schemas.upnp.org/upnp/1/0/"; ns=01\r\n
    SERVER: Linux/Unix 4.14.138.0 UPnP/1.0 RSSDP/1.0\r\n
    CONFIGID.UPNP.ORG: 1\r\n
    \r\n
```

BaronGreenback commented:
There are changes in how the system works - due to the way M$ implement ipv4 and ipv6 dual supported sockets.

Have also just come across v2 of ssdp which changes how the header work again! (No more OPT and XX-NLS)

PrplHaz4 commented:
> 
> 
> There are changes in how the system works - due to the way M$ implement ipv4 and ipv6 dual supported sockets.
> 
> Have also just come across v2 of ssdp which changes how the header work again! (No more OPT and XX-NLS)

It also looks like the NOTIFY messages are happening at 30min intervals now - not sure if that has to do with DLNA settings not saving/working (mine were set to 120, but now blank, and reset to 120 but still only see them at 30min intervals)...

BaronGreenback commented:
I know the reason - the web functionality is out of sync with the back end.

Will upload a migration fix later.

Keridos commented:
For reference (sent this in matrix troubleshhot room, too) https://gist.github.com/Keridos/c0abb3c1ab5259b7f6ae3d739f0f6b67
This happened when I tried to dlna play to my samsung tv. its profile worked in 10.5/10.6 and there seem to be two issues: the subtitlecodecnotsupported and the OOM error. Server had 2GB available, so it probable tried to allocate a lot of ram.

BaronGreenback commented:
@Keridos Think i've found the memory leak. It's in the latest unstable. Will patch this PR tomorrow when i'm on my dev mahine, and get it rebuilt.

BaronGreenback commented:
@Keridos This one has the bug fix included.

BaronGreenback commented:
@crobibero Could you update crobibero/jellyfin:3257 please?

crobibero commented:
updated `crobibero/jellyfin:3257` to a1e9986

Keridos commented:
Got this on trying to dlna play to my tv: https://gist.github.com/Keridos/8ead37692d0e543b38347a63892edfa2

Keridos commented:
@crobibero Could you please update the docker image again?

crobibero commented:
updated `crobibero/jellyfin:3257` to 13cf30a

Keridos commented:
@BaronGreenback in https://github.com/jellyfin/jellyfin/commit/13cf30a58f46d7c9a9df67b133b402e169e684f3 my webinterface stopped working. I tried with and without --published-server-url and set the correct network interface in system.xml via <LocalNetworkAddresses> without luck.

BaronGreenback commented:
@crobibero could you do the honors please?

BaronGreenback commented:
@Keridos Check the tracing options for the Dnla.xml file specified above for tracing parameters.

i suggest setting your logging.default.js to 
   "MinimumLevel": {
            "Default": "Warning",
          "Override": {
            "Microsoft": "Warning",
            "Microsoft.AspNetCore.Mvc" : "Warning",
            "Emby.Server.Implementations": "Warning",
            "Emby.Server.Implementations.Networking": "Debug",
            "Emby.Dlna.PlayTo.PlayToManager": "Debug",
            "Emby.Dlna": "Debug",
            "Emby.Server.Implementations.EntryPoints" : "Debug"

Should give you some idea as to what is happening.

Are you saying you need the loopback to be listened to for udp inbound traffic?

crobibero commented:
Updated `crobibero/jellyfin:3257` to 1c3d332

Keridos commented:
> @Keridos Check the tracing options for the Dnla.xml file specified above for tracing parameters.
> 
> i suggest setting your logging.default.js to
> "MinimumLevel": {
> "Default": "Warning",
> "Override": {
> "Microsoft": "Warning",
> "Microsoft.AspNetCore.Mvc" : "Warning",
> "Emby.Server.Implementations": "Warning",
> "Emby.Server.Implementations.Networking": "Debug",
> "Emby.Dlna.PlayTo.PlayToManager": "Debug",
> "Emby.Dlna": "Debug",
> "Emby.Server.Implementations.EntryPoints" : "Debug"
> 
> Should give you some idea as to what is happening.

Did that and enabled all the tracing options in the dlna log -> https://gist.github.com/Keridos/ec4ac225f27f874896d1eacba1080ae1


> Are you saying you need the loopback to be listened to for udp inbound traffic?

I do not know exactly but I think something like that.

BaronGreenback commented:
@crobibero Sorry twice in one day. Could you do the honors again please?

BaronGreenback commented:
@Keridos the image is getting rebuilt again. Could you repeat your test and send me the results please? It should now include the code that i need to discover why your tv doesn't like subscriptions. Precondition Failed can have multiple causes.

crobibero commented:
Updated `crobibero/jellyfin:3257` to 5fe974d

Keridos commented:
@BaronGreenback new logs: https://gist.github.com/Keridos/e18386c3b0762a82ee6932b653f2b63e
Though it doesnt look like the reason for the Precondition failed is in there.

BaronGreenback commented:
Could you check to see if you ran that on the latest code? Crobibero has just come back to say he's rebuilt it.

Keridos commented:
> 
> 
> Could you check to see if you ran that on the latest code? Crobibero has just come back to say he's rebuilt it.

Yeah, pretty sure it used the latest build, Waited until docker.hub showed the new build and then pulled again and recreated the container. We should maybe consider switching to matrix for communication for the time being, as this tends to get a bit "spammy" in here imho.

Keridos commented:
Just a small side note, I build the docker image now and then, too. it is available under keridos/jellyfin:3257
https://hub.docker.com/r/keridos/jellyfin/tags

PrplHaz4 commented:
@BaronGreenback seeing this as of fadf709:

```
[20:14:00] [ERR] SearchAsync failed.
System.Net.Sockets.SocketException (95): Operation not supported
   at System.Net.IPAddress.get_ScopeId()
   at Emby.Dlna.Net.SsdpServer.SendMulticastSSDP(Dictionary`2 values, String classification, IPAddress advertising, Nullable`1 sendCount) in /opt/jellyfin/Emby.Dlna/Net/SsdpServer.cs:line 163
   at Emby.Dlna.Net.SsdpPlayToLocator.OnBroadcastTimerCallback(Object state) in /opt/jellyfin/Emby.Dlna/Net/SsdpPlayToLocator.cs:line 166
```

BaronGreenback commented:
Thanks - Github at my end screwed up the merge just as i got this error - hence all the conflicting files.

Am manually going through them all.



Had just got IPv6 working lovely as well!

PrplHaz4 commented:
This is from 29bfc95 - I thought it might have to do with an old logging config but didn't seem to fix it...

```
[12:59:13] [ERR] Error creating Emby.Dlna.Main.DlnaEntryPoint
System.InvalidOperationException: Unable to resolve service for type 'Microsoft.Extensions.Logging.ILogger' while attempting to activate 'Emby.Dlna.Net.SsdpServer'.
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateArgumentCallSites(Type serviceType, Type implementationType, CallSiteChain callSiteChain, ParameterInfo[] parameters, Boolean throwIfCallSiteNotFound)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateConstructorCallSite(ResultCache lifetime, Type serviceType, Type implementationType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(ServiceDescriptor descriptor, Type serviceType, CallSiteChain callSiteChain, Int32 slot)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.TryCreateExact(Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.CreateCallSite(Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.<>c__DisplayClass7_0.<GetCallSite>b__0(Type type)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteFactory.GetCallSite(Type serviceType, CallSiteChain callSiteChain)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.CreateServiceAccessor(Type serviceType)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType, ServiceProviderEngineScope serviceProviderEngineScope)
   at Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngine.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(Type serviceType)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.ConstructorMatcher.CreateInstance(IServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.CreateInstance(IServiceProvider provider, Type instanceType, Object[] parameters)
   at Emby.Server.Implementations.ApplicationHost.CreateInstanceSafe(Type type) in /opt/jellyfin/Emby.Server.Implementations/ApplicationHost.cs:line 366
```

BaronGreenback commented:
Still sorting out the merge cock up. Just checking one last time before the re-upload

PrplHaz4 commented:
First impressions of cd672c5:
- Compiled OK
- Server discovery via VLC OK
- Server discovery via Device Sniffer OK

Issues
- LiveTV missing from VLC directory listings (may be unrelated to this since I know there were some plugin changes)
- Playback of some Movies in VLC slow/fails (also probably unrelated)

BaronGreenback commented:
Thanks for the feedback - just looking into the playback failing. Seeing it on PlayTo as well.

Keridos commented:
Having a weird issue that curl and my reverse proxy cannot access the server at port 8096 but it seems my firefox is able to do so.

Keridos commented:
https://gist.github.com/Keridos/c0c70730d51039496e7585b4c3d2f54f happened on trying to dlna playto with https://github.com/jellyfin/jellyfin/pull/3994/commits/2665b784dcba2d1381e62baf82415dff4499492b

Keridos commented:
Webui is partially broken, images are all blurred and it doesnt forward from root to /web/index.html
My TV still shows the same "unexpected" error. Heres the log: https://gist.github.com/Keridos/28284b553c61b556d3f4b1d8f6b8d9a0

BaronGreenback commented:
I've been manually updating the web and running it from a separate folder outside.
Looks like it's one bug down, a few to go!

BaronGreenback commented:
https://github.com/jellyfin/jellyfin/pull/4093

Keridos commented:
Webui still broken the same as in https://github.com/jellyfin/jellyfin/pull/3994#issuecomment-688539362
DLNA playback exiting on TV with same error as before.

Logs:
https://gist.github.com/Keridos/d5f02b88fbcf9e49b7fd0f2e06c36e3e on https://github.com/jellyfin/jellyfin/pull/3994/commits/86ade87ffbe7b004b4d23e45070a59bd04d7f4c9

PrplHaz4 commented:
As of 9e8e41d

Working
- DLNA Discovery (VLC-Windows, Device Sniffer, Device Validator)
- DLNA Movie playback/skipping (VLC-Windows)
- Web playback, DirectPlay, Transcode

Issues
- DLNA Discovery - Not Found (VLC-iOS, Samsung TV)
- LiveTV channels not listed in DLNA (VLC-Windows)
- LiveTV web playback failing on transcode with an HDHR:
```
[http @ 0x559b2df06b40] HTTP error 404 Not Found
http://127.0.0.1:8096/LiveTv/LiveStreamFiles/8a0b2b791bb14001952565e6ce956ebf/stream.ts: Server returned 404 Not Found
```

BaronGreenback commented:
@PrplHaz4 The http controller cannot find a live stream with the id provided. Haven't changed this code for a while.

I've added debugging to the HdHomerunUdpStream to give some pointers.

PrplHaz4 commented:
### ### 

> 
> 
> @PrplHaz4 The http controller cannot find a live stream with the id provided. Haven't changed this code for a while.
> 
> I've added debugging to the HdHomerunUdpStream to give some pointers.

I actually assumed the issue was with the server not listening on 127.0.0.1 - as I thought that had changed at some point?  Also, for some reason I think "newer" HDHRs use http streams not UDP (but could be wrong)...

Tried again with 80c7830 and I'm still getting the same 404 error:
```
[http @ 0x56362de46d80] HTTP error 404 Not Found
http://127.0.0.1:8096/LiveTv/LiveStreamFiles/67a0f5b68abd40c287e7419891fd7517/stream.ts: Server returned 404 Not Found
```

Here's the system log:
```
[13:41:16] [INF] Request starting HTTP/1.1 POST http://jftest2.mydomain.com/Items/cda10c0a-8b5c-5372-5d0b-b3aa3716cb2f/PlaybackInfo?UserId=65379ef7-bcf1-4db6-a1b6-446c558402c8&StartTimeTicks=0&IsPlayback=true&AutoOpenLiveStream=true&MaxStreamingBitrate=140000000 application/json 3187
[13:41:16] [INF] CORS policy execution successful.
[13:41:16] [DBG] AuthenticationScheme: CustomAuthentication was successfully authenticated.
[13:41:16] [DBG] AuthenticationScheme: CustomAuthentication was successfully authenticated.
[13:41:16] [INF] Authorization was successful.
[13:41:16] [INF] Executing endpoint 'Jellyfin.Api.Controllers.MediaInfoController.GetPostedPlaybackInfo (Jellyfin.Api)'
[13:41:16] [INF] Route matched with {action = "GetPostedPlaybackInfo", controller = "MediaInfo"}. Executing controller action with signature System.Threading.Tasks.Task`1[Microsoft.AspNetCore.Mvc.ActionResult`1[MediaBrowser.Model.MediaInfo.PlaybackInfoResponse]] GetPostedPlaybackInfo(System.Guid, System.Nullable`1[System.Guid], System.Nullable`1[System.Int64], System.Nullable`1[System.Int64], System.Nullable`1[System.Int32], System.Nullable`1[System.Int32], System.Nullable`1[System.Int32], System.String, System.String, Jellyfin.Api.Models.VideoDtos.DeviceProfileDto, Boolean, Boolean, Boolean, Boolean, Boolean, Boolean) on controller Jellyfin.Api.Controllers.MediaInfoController (Jellyfin.Api).
[13:41:16] [INF] GetPostedPlaybackInfo profile: {"Name": null, "Id": null, "Identification": null, "FriendlyName": null, "Manufacturer": null, "ManufacturerUrl": null, "ModelName": null, "ModelDescription": null, "ModelNumber": null, "ModelUrl": null, "SerialNumber": null, "EnableAlbumArtInDidl": false, "EnableSingleAlbumArtLimit": false, "EnableSingleSubtitleLimit": false, "SupportedMediaTypes": "Audio,Photo,Video", "UserId": null, "AlbumArtPn": null, "MaxAlbumArtWidth": 0, "MaxAlbumArtHeight": 0, "MaxIconWidth": null, "MaxIconHeight": null, "MaxStreamingBitrate": 120000000, "MaxStaticBitrate": 100000000, "MusicStreamingTranscodingBitrate": 192000, "MaxStaticMusicBitrate": null, "SonyAggregationFlags": null, "ProtocolInfo": null, "TimelineOffsetSeconds": 0, "RequiresPlainVideoItems": false, "RequiresPlainFolders": false, "EnableMSMediaReceiverRegistrar": false, "IgnoreTranscodeByteRangeRequests": false, "XmlRootAttributes": [], "DirectPlayProfiles": [{"Container": "webm", "AudioCodec": "vorbis,opus", "VideoCodec": "vp8,vp9,av1", "Type": "Video", "$type": "DirectPlayProfile"}, {"Container": "mp4,m4v", "AudioCodec": "mp3,aac,opus,flac,vorbis", "VideoCodec": "h264,vp8,vp9,av1", "Type": "Video", "$type": "DirectPlayProfile"}, {"Container": "opus", "AudioCodec": null, "VideoCodec": null, "Type": "Audio", "$type": "DirectPlayProfile"}, {"Container": "mp3", "AudioCodec": "mp3", "VideoCodec": null, "Type": "Audio", "$type": "DirectPlayProfile"}, {"Container": "aac", "AudioCodec": null, "VideoCodec": null, "Type": "Audio", "$type": "DirectPlayProfile"}, {"Container": "m4a,m4b", "AudioCodec": "aac", "VideoCodec": null, "Type": "Audio", "$type": "DirectPlayProfile"}, {"Container": "flac", "AudioCodec": null, "VideoCodec": null, "Type": "Audio", "$type": "DirectPlayProfile"}, {"Container": "webma,webm", "AudioCodec": null, "VideoCodec": null, "Type": "Audio", "$type": "DirectPlayProfile"}, {"Container": "wav", "AudioCodec": null, "VideoCodec": null, "Type": "Audio", "$type": "DirectPlayProfile"}, {"Container": "ogg", "AudioCodec": null, "VideoCodec": null, "Type": "Audio", "$type": "DirectPlayProfile"}], "TranscodingProfiles": [{"Container": "ts", "Type": "Audio", "VideoCodec": null, "AudioCodec": "aac", "Protocol": "hls", "EstimateContentLength": false, "EnableMpegtsM2TsMode": false, "TranscodeSeekInfo": "Auto", "CopyTimestamps": false, "Context": "Streaming", "EnableSubtitlesInManifest": false, "MaxAudioChannels": "2", "MinSegments": 1, "SegmentLength": 0, "BreakOnNonKeyFrames": true, "$type": "TranscodingProfile"}, {"Container": "aac", "Type": "Audio", "VideoCodec": null, "AudioCodec": "aac", "Protocol": "http", "EstimateContentLength": false, "EnableMpegtsM2TsMode": false, "TranscodeSeekInfo": "Auto", "CopyTimestamps": false, "Context": "Streaming", "EnableSubtitlesInManifest": false, "MaxAudioChannels": "2", "MinSegments": 0, "SegmentLength": 0, "BreakOnNonKeyFrames": false, "$type": "TranscodingProfile"}, {"Container": "mp3", "Type": "Audio", "VideoCodec": null, "AudioCodec": "mp3", "Protocol": "http", "EstimateContentLength": false, "EnableMpegtsM2TsMode": false, "TranscodeSeekInfo": "Auto", "CopyTimestamps": false, "Context": "Streaming", "EnableSubtitlesInManifest": false, "MaxAudioChannels": "2", "MinSegments": 0, "SegmentLength": 0, "BreakOnNonKeyFrames": false, "$type": "TranscodingProfile"}, {"Container": "opus", "Type": "Audio", "VideoCodec": null, "AudioCodec": "opus", "Protocol": "http", "EstimateContentLength": false, "EnableMpegtsM2TsMode": false, "TranscodeSeekInfo": "Auto", "CopyTimestamps": false, "Context": "Streaming", "EnableSubtitlesInManifest": false, "MaxAudioChannels": "2", "MinSegments": 0, "SegmentLength": 0, "BreakOnNonKeyFrames": false, "$type": "TranscodingProfile"}, {"Container": "wav", "Type": "Audio", "VideoCodec": null, "AudioCodec": "wav", "Protocol": "http", "EstimateContentLength": false, "EnableMpegtsM2TsMode": false, "TranscodeSeekInfo": "Auto", "CopyTimestamps": false, "Context": "Streaming", "EnableSubtitlesInManifest": false, "MaxAudioChannels": "2", "MinSegments": 0, "SegmentLength": 0, "BreakOnNonKeyFrames": false, "$type": "TranscodingProfile"}, {"Container": "opus", "Type": "Audio", "VideoCodec": null, "AudioCodec": "opus", "Protocol": "http", "EstimateContentLength": false, "EnableMpegtsM2TsMode": false, "TranscodeSeekInfo": "Auto", "CopyTimestamps": false, "Context": "Static", "EnableSubtitlesInManifest": false, "MaxAudioChannels": "2", "MinSegments": 0, "SegmentLength": 0, "BreakOnNonKeyFrames": false, "$type": "TranscodingProfile"}, {"Container": "mp3", "Type": "Audio", "VideoCodec": null, "AudioCodec": "mp3", "Protocol": "http", "EstimateContentLength": false, "EnableMpegtsM2TsMode": false, "TranscodeSeekInfo": "Auto", "CopyTimestamps": false, "Context": "Static", "EnableSubtitlesInManifest": false, "MaxAudioChannels": "2", "MinSegments": 0, "SegmentLength": 0, "BreakOnNonKeyFrames": false, "$type": "TranscodingProfile"}, {"Container": "aac", "Type": "Audio", "VideoCodec": null, "AudioCodec": "aac", "Protocol": "http", "EstimateContentLength": false, "EnableMpegtsM2TsMode": false, "TranscodeSeekInfo": "Auto", "CopyTimestamps": false, "Context": "Static", "EnableSubtitlesInManifest": false, "MaxAudioChannels": "2", "MinSegments": 0, "SegmentLength": 0, "BreakOnNonKeyFrames": false, "$type": "TranscodingProfile"}, {"Container": "wav", "Type": "Audio", "VideoCodec": null, "AudioCodec": "wav", "Protocol": "http", "EstimateContentLength": false, "EnableMpegtsM2TsMode": false, "TranscodeSeekInfo": "Auto", "CopyTimestamps": false, "Context": "Static", "EnableSubtitlesInManifest": false, "MaxAudioChannels": "2", "MinSegments": 0, "SegmentLength": 0, "BreakOnNonKeyFrames": false, "$type": "TranscodingProfile"}, {"Container": "ts", "Type": "Video", "VideoCodec": "h264", "AudioCodec": "mp3,aac", "Protocol": "hls", "EstimateContentLength": false, "EnableMpegtsM2TsMode": false, "TranscodeSeekInfo": "Auto", "CopyTimestamps": false, "Context": "Streaming", "EnableSubtitlesInManifest": false, "MaxAudioChannels": "2", "MinSegments": 1, "SegmentLength": 0, "BreakOnNonKeyFrames": true, "$type": "TranscodingProfile"}, {"Container": "webm", "Type": "Video", "VideoCodec": "vpx", "AudioCodec": "vorbis", "Protocol": "http", "EstimateContentLength": false, "EnableMpegtsM2TsMode": false, "TranscodeSeekInfo": "Auto", "CopyTimestamps": false, "Context": "Streaming", "EnableSubtitlesInManifest": false, "MaxAudioChannels": "2", "MinSegments": 0, "SegmentLength": 0, "BreakOnNonKeyFrames": false, "$type": "TranscodingProfile"}, {"Container": "mp4", "Type": "Video", "VideoCodec": "h264", "AudioCodec": "mp3,aac,opus,flac,vorbis", "Protocol": "http", "EstimateContentLength": false, "EnableMpegtsM2TsMode": false, "TranscodeSeekInfo": "Auto", "CopyTimestamps": false, "Context": "Static", "EnableSubtitlesInManifest": false, "MaxAudioChannels": null, "MinSegments": 0, "SegmentLength": 0, "BreakOnNonKeyFrames": false, "$type": "TranscodingProfile"}], "ContainerProfiles": [], "CodecProfiles": [{"Type": "VideoAudio", "Conditions": [{"Condition": "Equals", "Property": "IsSecondaryAudio", "Value": "false", "IsRequired": false, "$type": "ProfileCondition"}], "ApplyConditions": [], "Codec": "aac", "Container": null, "$type": "CodecProfile"}, {"Type": "VideoAudio", "Conditions": [{"Condition": "Equals", "Property": "IsSecondaryAudio", "Value": "false", "IsRequired": false, "$type": "ProfileCondition"}], "ApplyConditions": [], "Codec": null, "Container": null, "$type": "CodecProfile"}, {"Type": "Video", "Conditions": [{"Condition": "NotEquals", "Property": "IsAnamorphic", "Value": "true", "IsRequired": false, "$type": "ProfileCondition"}, {"Condition": "EqualsAny", "Property": "VideoProfile", "Value": "high|main|baseline|constrained baseline", "IsRequired": false, "$type": "ProfileCondition"}, {"Condition": "LessThanEqual", "Property": "VideoLevel", "Value": "51", "IsRequired": false, "$type": "ProfileCondition"}, {"Condition": "NotEquals", "Property": "IsInterlaced", "Value": "true", "IsRequired": false, "$type": "ProfileCondition"}], "ApplyConditions": [], "Codec": "h264", "Container": null, "$type": "CodecProfile"}], "ResponseProfiles": [{"Container": "m4v", "AudioCodec": null, "VideoCodec": null, "Type": "Video", "OrgPn": null, "MimeType": "video/mp4", "Conditions": [], "$type": "ResponseProfile"}], "SubtitleProfiles": [{"Format": "vtt", "Method": "External", "DidlMode": null, "Language": null, "Container": null, "$type": "SubtitleProfile"}, {"Format": "ass", "Method": "External", "DidlMode": null, "Language": null, "Container": null, "$type": "SubtitleProfile"}, {"Format": "ssa", "Method": "External", "DidlMode": null, "Language": null, "Container": null, "$type": "SubtitleProfile"}], "$type": "DeviceProfile"}
[13:41:17] [INF] Start processing HTTP request GET http://192.168.4.37/discover.json
[13:41:17] [INF] Sending HTTP request GET http://192.168.4.37/discover.json
[13:41:17] [INF] Received HTTP response after 7.2136ms - OK
[13:41:17] [INF] End processing HTTP request after 8.5949ms - OK
[13:41:17] [INF] Start processing HTTP request GET http://192.168.4.37/lineup.json
[13:41:17] [INF] Sending HTTP request GET http://192.168.4.37/lineup.json
[13:41:17] [INF] Received HTTP response after 4.2824ms - OK
[13:41:17] [INF] End processing HTTP request after 5.2532ms - OK
[13:41:17] [DBG] GetBindInterface: Souce: False, External: False:
[13:41:17] [DBG] 255.255.255.255/32: GetBindInterface: Matched first internal interface. 192.168.4.14
[13:41:17] [DBG] MediaSources: [{"Protocol": "Udp", "Id": "native_b23ebc9442ec90ed5f591ef1683972b1_81909ee7ce0a737df7c0f6729bb5b7b6", "Path": "http://192.168.4.37", "EncoderPath": null, "EncoderProtocol": null, "Type": "Default", "Container": "ts", "Size": null, "Name": null, "IsRemote": false, "ETag": null, "RunTimeTicks": null, "ReadAtNativeFramerate": false, "IgnoreDts": true, "IgnoreIndex": false, "GenPtsInput": false, "SupportsTranscoding": true, "SupportsDirectStream": true, "SupportsDirectPlay": false, "IsInfiniteStream": true, "RequiresOpening": true, "OpenToken": "LiveTvChannel_cda10c0a8b5c53725d0bb3aa3716cb2f_native_b23ebc9442ec90ed5f591ef1683972b1_81909ee7ce0a737df7c0f6729bb5b7b6", "RequiresClosing": true, "LiveStreamId": null, "BufferMs": 0, "RequiresLooping": false, "SupportsProbing": true, "VideoType": null, "IsoType": null, "Video3DFormat": null, "MediaStreams": [{"Codec": "mpeg2video", "CodecTag": null, "Language": null, "ColorRange": null, "ColorSpace": null, "ColorTransfer": null, "ColorPrimaries": null, "Comment": null, "TimeBase": null, "CodecTimeBase": null, "Title": null, "VideoRange": "SDR", "localizedUndefined": null, "localizedDefault": null, "localizedForced": null, "DisplayTitle": "1080i MPEG2VIDEO", "NalLengthSize": null, "IsInterlaced": true, "IsAVC": null, "ChannelLayout": null, "BitRate": 15000000, "BitDepth": null, "RefFrames": null, "PacketLength": null, "Channels": null, "SampleRate": null, "IsDefault": false, "IsForced": false, "Height": 1080, "Width": 1920, "AverageFrameRate": null, "RealFrameRate": null, "Profile": null, "Type": "Video", "AspectRatio": null, "Index": -1, "Score": null, "IsExternal": false, "DeliveryMethod": null, "DeliveryUrl": null, "IsExternalUrl": null, "IsTextSubtitleStream": false, "SupportsExternalStream": false, "Path": null, "PixelFormat": null, "Level": null, "IsAnamorphic": null, "$type": "MediaStream"}, {"Codec": "AC3", "CodecTag": null, "Language": null, "ColorRange": null, "ColorSpace": null, "ColorTransfer": null, "ColorPrimaries": null, "Comment": null, "TimeBase": null, "CodecTimeBase": null, "Title": null, "VideoRange": null, "localizedUndefined": null, "localizedDefault": null, "localizedForced": null, "DisplayTitle": "Dolby Digital", "NalLengthSize": null, "IsInterlaced": false, "IsAVC": null, "ChannelLayout": null, "BitRate": 448000, "BitDepth": null, "RefFrames": null, "PacketLength": null, "Channels": null, "SampleRate": null, "IsDefault": false, "IsForced": false, "Height": null, "Width": null, "AverageFrameRate": null, "RealFrameRate": null, "Profile": null, "Type": "Audio", "AspectRatio": null, "Index": -1, "Score": null, "IsExternal": false, "DeliveryMethod": null, "DeliveryUrl": null, "IsExternalUrl": null, "IsTextSubtitleStream": false, "SupportsExternalStream": false, "Path": null, "PixelFormat": null, "Level": null, "IsAnamorphic": null, "$type": "MediaStream"}], "MediaAttachments": [], "Formats": [], "Bitrate": 15448000, "Timestamp": null, "RequiredHttpHeaders": {}, "TranscodingUrl": null, "TranscodingSubProtocol": null, "TranscodingContainer": null, "AnalyzeDurationMs": null, "TranscodeReasons": null, "DefaultAudioStreamIndex": null, "DefaultSubtitleStreamIndex": null, "VideoStream": {"Codec": "mpeg2video", "CodecTag": null, "Language": null, "ColorRange": null, "ColorSpace": null, "ColorTransfer": null, "ColorPrimaries": null, "Comment": null, "TimeBase": null, "CodecTimeBase": null, "Title": null, "VideoRange": "SDR", "localizedUndefined": null, "localizedDefault": null, "localizedForced": null, "DisplayTitle": "1080i MPEG2VIDEO", "NalLengthSize": null, "IsInterlaced": true, "IsAVC": null, "ChannelLayout": null, "BitRate": 15000000, "BitDepth": null, "RefFrames": null, "PacketLength": null, "Channels": null, "SampleRate": null, "IsDefault": false, "IsForced": false, "Height": 1080, "Width": 1920, "AverageFrameRate": null, "RealFrameRate": null, "Profile": null, "Type": "Video", "AspectRatio": null, "Index": -1, "Score": null, "IsExternal": false, "DeliveryMethod": null, "DeliveryUrl": null, "IsExternalUrl": null, "IsTextSubtitleStream": false, "SupportsExternalStream": false, "Path": null, "PixelFormat": null, "Level": null, "IsAnamorphic": null, "$type": "MediaStream"}, "$type": "MediaSourceInfo"}]
[13:41:17] [INF] User policy for jjh. EnablePlaybackRemuxing: True EnableVideoPlaybackTranscoding: True EnableAudioPlaybackTranscoding: True
[13:41:17] [INF] Profile: Unknown Profile, Path: http://192.168.4.37, isEligibleForDirectPlay: True, isEligibleForDirectStream: True
[13:41:17] [INF] Profile: Unknown Profile, No direct play profiles found for Path: http://192.168.4.37
[13:41:17] [INF] Opening channel stream from Emby, external channel Id: hdhr_2.1
[13:41:17] [INF] Streaming Channel hdhr_2.1
[13:41:17] [INF] GetChannelStream: channel id: hdhr_2.1. stream id: native_b23ebc9442ec90ed5f591ef1683972b1_81909ee7ce0a737df7c0f6729bb5b7b6 profile: native
[13:41:17] [INF] Opening SharedHttpStream Live stream from http://192.168.4.37:5004/auto/v2.1
[13:41:17] [INF] Start processing HTTP request GET http://192.168.4.37:5004/auto/v2.1
[13:41:17] [INF] Sending HTTP request GET http://192.168.4.37:5004/auto/v2.1
[13:41:17] [INF] Received HTTP response after 193.8865ms - OK
[13:41:17] [INF] End processing HTTP request after 194.2991ms - OK
[13:41:17] [INF] Beginning SharedHttpStream stream to /config/transcoding-temp/transcodes/6a8a84fa15ea429eaba7540612dd0ee4.ts
[13:41:17] [INF] Live stream opened after 204.9535ms
[13:41:17] [INF] Returning mediasource streamId native_b23ebc9442ec90ed5f591ef1683972b1_81909ee7ce0a737df7c0f6729bb5b7b6, mediaSource.Id native_b23ebc9442ec90ed5f591ef1683972b1_81909ee7ce0a737df7c0f6729bb5b7b6, mediaSource.LiveStreamId null
[13:41:17] [INF] Waiting 3000ms before probing the live stream
[13:41:17] [INF] Deleting temp files ["/config/transcoding-temp/transcodes/6a8a84fa15ea429eaba7540612dd0ee4.ts"]
[13:41:20] [INF] /usr/lib/jellyfin-ffmpeg/ffprobe -analyzeduration 3000000 -i "http://127.0.0.1:8096/LiveTv/LiveStreamFiles/6a8a84fa15ea429eaba7540612dd0ee4/stream.ts" -threads 0 -v warning -print_format json -show_streams -show_format
[13:41:20] [DBG] Starting ffprobe with args -analyzeduration 3000000 -i "http://127.0.0.1:8096/LiveTv/LiveStreamFiles/6a8a84fa15ea429eaba7540612dd0ee4/stream.ts" -threads 0 -v warning -print_format json -show_streams -show_format
[13:41:20] [INF] Request starting HTTP/1.1 GET http://127.0.0.1:8096/LiveTv/LiveStreamFiles/6a8a84fa15ea429eaba7540612dd0ee4/stream.ts  
[13:41:20] [DBG] AuthenticationScheme: CustomAuthentication was not authenticated.
[13:41:20] [INF] Executing endpoint 'Jellyfin.Api.Controllers.LiveTvController.GetLiveStreamFile (Jellyfin.Api)'
[13:41:20] [INF] Route matched with {action = "GetLiveStreamFile", controller = "LiveTv"}. Executing controller action with signature System.Threading.Tasks.Task`1[Microsoft.AspNetCore.Mvc.ActionResult] GetLiveStreamFile(System.String, System.String) on controller Jellyfin.Api.Controllers.LiveTvController (Jellyfin.Api).
[13:41:20] [INF] Executed action Jellyfin.Api.Controllers.LiveTvController.GetLiveStreamFile (Jellyfin.Api) in 10.2717ms
[13:41:20] [INF] Executed endpoint 'Jellyfin.Api.Controllers.LiveTvController.GetLiveStreamFile (Jellyfin.Api)'
[13:41:20] [ERR] Error processing request: Could not find file '/config/transcoding-temp/transcodes/6a8a84fa15ea429eaba7540612dd0ee4.ts'. URL GET /LiveTv/LiveStreamFiles/6a8a84fa15ea429eaba7540612dd0ee4/stream.ts.
[13:41:20] [INF] Request finished in 18.1183ms 404 text/plain
[http @ 0x564347ac9c00] HTTP error 404 Not Found
http://127.0.0.1:8096/LiveTv/LiveStreamFiles/6a8a84fa15ea429eaba7540612dd0ee4/stream.ts: Server returned 404 Not Found
[13:41:20] [ERR] Error probing live tv stream
System.Exception: ffprobe failed - streams and format are both null.
   at MediaBrowser.MediaEncoding.Encoder.MediaEncoder.GetMediaInfoInternal(String inputPath, String primaryPath, MediaProtocol protocol, Boolean extractChapters, String probeSizeArgument, Boolean isAudio, Nullable`1 videoType, Boolean forceEnableLogging, CancellationToken cancellationToken) in /opt/jellyfin/MediaBrowser.MediaEncoding/Encoder/MediaEncoder.cs:line 437
   at Emby.Server.Implementations.Library.LiveStreamHelper.AddMediaInfoWithProbe(MediaSourceInfo mediaSource, Boolean isAudio, String cacheKey, Boolean addProbeDelay, CancellationToken cancellationToken) in /opt/jellyfin/Emby.Server.Implementations/Library/LiveStreamHelper.cs:line 74
   at Emby.Server.Implementations.Library.MediaSourceManager.OpenLiveStreamInternal(LiveStreamRequest request, CancellationToken cancellationToken) in /opt/jellyfin/Emby.Server.Implementations/Library/MediaSourceManager.cs:line 506
[13:41:20] [INF] Live stream opened: {"Protocol":"Http","Id":"native_b23ebc9442ec90ed5f591ef1683972b1_81909ee7ce0a737df7c0f6729bb5b7b6","Path":"http://127.0.0.1:8096/LiveTv/LiveStreamFiles/6a8a84fa15ea429eaba7540612dd0ee4/stream.ts","Type":"Default","Container":"ts","IsRemote":false,"ReadAtNativeFramerate":false,"IgnoreDts":true,"IgnoreIndex":false,"GenPtsInput":false,"SupportsTranscoding":true,"SupportsDirectStream":true,"SupportsDirectPlay":false,"IsInfiniteStream":true,"RequiresOpening":true,"RequiresClosing":true,"LiveStreamId":"a17c75760a04e99b68cf766e11316e1c_09efa0d56b934a82adec00a87b837fb0_native_b23ebc9442ec90ed5f591ef1683972b1_81909ee7ce0a737df7c0f6729bb5b7b6","BufferMs":0,"RequiresLooping":false,"SupportsProbing":true,"MediaStreams":[{"Codec":"mpeg2video","VideoRange":"SDR","DisplayTitle":"1080i MPEG2VIDEO","IsInterlaced":true,"BitRate":15000000,"IsDefault":false,"IsForced":false,"Height":1080,"Width":1920,"Type":"Video","Index":-1,"IsExternal":false,"IsTextSubtitleStream":false,"SupportsExternalStream":false},{"Codec":"AC3","DisplayTitle":"Dolby Digital","IsInterlaced":false,"BitRate":448000,"IsDefault":false,"IsForced":false,"Type":"Audio","Index":-1,"IsExternal":false,"IsTextSubtitleStream":false,"SupportsExternalStream":false}],"MediaAttachments":[],"Formats":[],"Bitrate":15448000,"RequiredHttpHeaders":{},"AnalyzeDurationMs":3000}
[13:41:20] [INF] User policy for jjh. EnablePlaybackRemuxing: True EnableVideoPlaybackTranscoding: True EnableAudioPlaybackTranscoding: True
[13:41:20] [INF] Profile: Unknown Profile, Path: http://127.0.0.1:8096/LiveTv/LiveStreamFiles/6a8a84fa15ea429eaba7540612dd0ee4/stream.ts, isEligibleForDirectPlay: True, isEligibleForDirectStream: True
[13:41:20] [INF] Profile: Unknown Profile, No direct play profiles found for Path: http://127.0.0.1:8096/LiveTv/LiveStreamFiles/6a8a84fa15ea429eaba7540612dd0ee4/stream.ts
[13:41:20] [INF] Profile: Unknown Profile, Path: http://127.0.0.1:8096/LiveTv/LiveStreamFiles/6a8a84fa15ea429eaba7540612dd0ee4/stream.ts, isEligibleForDirectPlay: True, isEligibleForDirectStream: True
[13:41:20] [INF] Profile: Unknown Profile, No direct play profiles found for Path: http://127.0.0.1:8096/LiveTv/LiveStreamFiles/6a8a84fa15ea429eaba7540612dd0ee4/stream.ts
[13:41:20] [INF] Executing ObjectResult, writing value of type 'MediaBrowser.Model.MediaInfo.PlaybackInfoResponse'.
[13:41:20] [WRN] Slow HTTP Response from http://jftest2.mydomain.com/Items/cda10c0a-8b5c-5372-5d0b-b3aa3716cb2f/PlaybackInfo?UserId=65379ef7-bcf1-4db6-a1b6-446c558402c8&StartTimeTicks=0&IsPlayback=true&AutoOpenLiveStream=true&MaxStreamingBitrate=140000000 to 172.23.0.2 in 0:00:03.7789513 with Status Code 200
[13:41:20] [INF] Executed action Jellyfin.Api.Controllers.MediaInfoController.GetPostedPlaybackInfo (Jellyfin.Api) in 3744.8654ms
[13:41:20] [INF] Executed endpoint 'Jellyfin.Api.Controllers.MediaInfoController.GetPostedPlaybackInfo (Jellyfin.Api)'
[13:41:20] [INF] Request finished in 3780.3586ms 200 application/json; charset=utf-8
```

PrplHaz4 commented:
@BaronGreenback sorry for the noise - I confirmed this LiveTV 404 error exists on master @ b76d4ba454c43cc6293b733f2bba363e34a9bf5e so is not a result of this PR.

hariseldon99 commented:
Nobody's reviewing these?

BaronGreenback commented:
This is end result - and is for testing. It's continually updated with the latest master.

As its such a large PR - it is being broken down into smaller PR's - each one submitted separately.

BaronGreenback commented:
Added some more testing units.

Also implemented the ability to mock network interface configurations. 
With this ability, it should be possible to replicate user setups, in order to trace problems.

BaronGreenback commented:
Moved all the udp coding into the external library. 

This will substantially reduce the amount of code to check in the overall PR.

PrplHaz4 commented:
> 
> It's not compiling as i was a bit quick - the NuGet package wasn't published.

I got this error when building:
```
PlayTo/SsdpPlayToLocator.cs(7,1): error SA1210: Using directives should be ordered alphabetically by the namespaces [/opt/jellyfin/Emby.Dlna/Emby.Dlna.csproj]
```

Moving `Ssdp` below `SSDP` seems to have fixed it 🥇 
```
using NetworkCollection.SSDP;
using NetworkCollection.Ssdp;
```

BaronGreenback commented:
Argghh! - package 1.0.4 on it's way.

BaronGreenback commented:
Backup your settings before running. This version splits the network config into network.xml. 

Without the web change implemented - changes in the web won't do anything.

BaronGreenback commented:
Relinked code back into nuget.

BaronGreenback commented:
Removed NetCollection class as requested by @cvium. Functionality is now provided via helper classes.

PrplHaz4 commented:
597af61 Build is failing but fixed with these:

- Add `using Microsoft.AspNetCore.Http;` to `IServerApplicationHost.cs` (removed in #4247)
- Remove duplicate `[Produces(MediaTypeNames.Text.Xml)]` on line 197 of `DlnaServerController.cs`

BaronGreenback commented:
Closing this one down.

Will open a new test based upon plugins.
